app/controllers/declaration/ProcedureCodesController.scala

```diff
@@ -126,15 +126,27 @@ class ProcedureCodesController @Inject()(
   private def updateCache(itemId: String, updatedProcedureCodes: ProcedureCodesData)(
     implicit r: JourneyRequest[AnyContent]
   ): Future[Option[ExportsDeclaration]] = {
+
+    def clearDataForProcedureCode(code: String, itemId: String, model: ExportsDeclaration) = {
+      val step1Model =
+        if (!ProcedureCodesData.osrProcedureCodes.contains(code))
+          model.updatedItem(itemId, item => item.copy(fiscalInformation = None, additionalFiscalReferencesData = None))
+        else model
+
+      val step2Model =
+        if (r.isType(DeclarationType.CLEARANCE) && ProcedureCodesData.eicrProcedureCodes.contains(code))
+          step1Model.updatedItem(itemId, item => item.copy(packageInformation = None))
+        else step1Model
+
+      step2Model
```

Comment:
I'd change this into a single "if" statement with 3 possible outcomes. Something like this:
`if (r.isType(DeclarationType.CLEARANCE) && ProcedureCodesData.eicrProcedureCodes.contains(code))
model.updatedItem(itemId, item => item.copy(fiscalInformation = None, additionalFiscalReferencesData = None, packageInformation = None))
else if (!ProcedureCodesData.osrProcedureCodes.contains(code))
model.updatedItem(itemId, item => item.copy(fiscalInformation = None, additionalFiscalReferencesData = None))
else
model`
It took me a while to figure out what and why is it done in two steps.

Reply:
but the problem here is that this must be in separate ifs. There can be a situation when two updates must be done in the same method

Reply:
That approach would fail to make both changes.  In other words, it is possible for both of the "if" statements to be true.  I.e. a code of "0019" is both equal to "0019" and not equal to "1042".
There is a reason for this mess.

Reply:
Ok, just read the long discussion under Paulo's comment. I agree that we can think about some other way of pipelining operations in such scenario.